agent/connect/ca/provider_vault.go

```diff
@@ -182,20 +156,88 @@ func (v *VaultProvider) GenerateIntermediate() (string, error) {
 	}
```

Comment:
Can we add `no_store` to the role we create here? https://www.vaultproject.io/api/secret/pki/index.html#no_store

We never rely on Vault to remember certificates and we'll deal with revocation ourselves later. Currently every cert we issue gets stored in Vault until it expires which is unnecessary write load on Vault and results in lots of pointless storage (which might well be hitting Consul KV!).

Issues like #4475 highlight this. It's orthoganol to your changes here but maybe a good time to add that while we are touching this and testing it etc?